x/bank/client/cli/cli_test.go

```diff
@@ -0,0 +1,157 @@
+			[]string{
+				val.Address.String(),
+				val.Address.String(),
+				sdk.NewCoins(
+					sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10)),
+					sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)),
+				).String(),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+				fmt.Sprintf("--%s=true", flags.FlagGenerateOnly),
+			},
+			&sdk.TxResponse{},
+			0,
+			false,
+		},
+		{
+			"valid transaction",
+			[]string{
+				val.ValAddress.String(),
+				val.ValAddress.String(),
+				sdk.NewCoins(
+					sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10)),
+					sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)),
+				).String(),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			&sdk.TxResponse{},
+			0,
+			false,
+		},
+		{
+			"not enough fees",
+			[]string{
+				val.ValAddress.String(),
+				val.ValAddress.String(),
+				sdk.NewCoins(
+					sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10)),
+					sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)),
+				).String(),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1))).String()),
+			},
+			&sdk.TxResponse{},
+			sdkerrors.ErrInsufficientFee.ABCICode(),
+			false,
+		},
+		{
+			"not enough gas",
+			[]string{
+				val.ValAddress.String(),
+				val.ValAddress.String(),
+				sdk.NewCoins(
+					sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10)),
+					sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)),
+				).String(),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+				"--gas=10",
+			},
+			&sdk.TxResponse{},
+			sdkerrors.ErrOutOfGas.ABCICode(),
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			clientCtx := val.ClientCtx.WithOutput(buf)
+
+			buf.Reset()
+
+			cmd := cli.LockedSendTxCmd()
+			cmd.SetErr(buf)
+			cmd.SetOut(buf)
+			cmd.SetArgs(tc.args)
+
+			err := cmd.Execute()
+			if tc.expectErr {
+				s.Require().Error(err)
+				fmt.Println(err)
+				fmt.Println(cmd)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NoError(clientCtx.JSONCodec.UnmarshalJSON(buf.Bytes(), tc.respType), buf.String())
+
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code)
```

Comment:
What are we testing here? it doesn't really make sense what we're trying to test